.github/workflows/release.yml

```diff
@@ -28,39 +28,57 @@ jobs:
             artifact: fixpoint-x86_64-apple-darwin
           - os: macos-14
             artifact: fixpoint-aarch64-apple-darwin
+          - os: windows-2025
+            artifact: fixpoint-x86_64-pc-windows
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
 
       - name: Set up Haskell
+        id: setup-haskell
         uses: haskell-actions/setup@v2
         with:
           ghc-version: ${{ env.GHC_VERSION }}
           cabal-version: ${{ env.CABAL_VERSION }}
 
-      # Use cache in PRs, but do a clean build before releases
-      - name: Cache ~/.cabal/packages, ~/.cabal/store and dist-newstyle
+      - name: Freeze Cabal plan
+        run: |
+          cabal update
+          cabal freeze
```

Comment:
Note that the documentation of haskell-actions/setup provides [examples](https://github.com/haskell-actions/setup/blob/main/docs/examples.md#model-cabal-workflow-with-caching) with `plan.json` instead of `cabal freeze`. `plan.json` does seem to be more precise.

Reply:
I don't have strong opinions there. Caching dependencies allows to save the cache if the build of the local packages fails.

Caching the full build might save some extra work as long as incremental builds are reliable. I haven't needed to clean caches more than once or twice in many years of using Haskell.